ReSwift/CoreTypes/Store.swift

```diff
@@ -17,14 +17,26 @@ import Foundation
  */
 public class Store<State: StateType>: StoreType {
 
+    typealias SubscriptionType = Subscription<State>
+
     // TODO: Setter should not be public; need way for store enhancers to modify appState anyway
 
     /*private (set)*/ public var state: State! {
         didSet {
-            subscriptions.forEach {
+            var indexesToRemove: [Int] = []
+            for (index, subscription) in subscriptions.enumerate() {
+                guard let subscriber = subscription.subscriber else {
+                    indexesToRemove.append(index)
+                    return
```

Comment:
If you like we can look into some performance test, but the operation
should be extremely cheap for all use cases I can think of. I'm also open
to other suggestions :) Ideally though I would like to treat these to tasks
separately to make the code easier to understand and to eliminate the risk
of subtle changes that could cause bugs.

--
Benjamin Encz

Find me on twitter: @benjaminencz <https://twitter.com/benjaminencz>

On Thu, Feb 4, 2016 at 1:46 PM, Virgilio Favero Neto <
notifications@github.com> wrote:

> In ReSwift/CoreTypes/Store.swift
> <https://github.com/ReSwift/ReSwift/pull/70#discussion_r51942915>:
>
> >      // TODO: Setter should not be public; need way for store enhancers to modify appState anyway
> >
> >      /*private (set)*/ public var state: State! {
> >          didSet {
> > -            subscriptions.forEach {
> > +            var indexesToRemove: [Int] = []
> > +            for (index, subscription) in subscriptions.enumerate() {
> > +                guard let subscriber = subscription.subscriber else {
> > +                    indexesToRemove.append(index)
> > +                    return
>
> @Ben-G <https://github.com/Ben-G>, I like how the code is concise with
> this approach, although I'm not sure if the best approach here is to run
> the filter on every attempt to notify the subscribers.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ReSwift/ReSwift/pull/70/files#r51942915>.
>